### DISCUSSION TITLE
Action 云编译 linux 部分存在问题

### DISCUSSION THREAD

**Discussion by Larvan2** [tw93/Pake#315] [Q&A] [answered]
使用[action](https://github.com/tw93/Pake/wiki/GitHub-Actions-%E5%9C%A8%E7%BA%BF%E7%BC%96%E8%AF%91%E5%A4%9A%E7%B3%BB%E7%BB%9F%E7%89%88%E6%9C%AC)编译时，linux编译结果如图所示，文件名与实际不符导致后面upload不了。 
![image](https://user-images.githubusercontent.com/78135608/219847348-98c88bc5-dc60-4f05-8edf-eb3df6334605.png)

[配置](https://github.com/Larvan2/Yacd-Desktop/tree/master/src-tauri)如下：
```json
{
  "tauri": {
    "bundle": {
      "icon": [
        "png/yacd_256.ico",
        "png/yacd_512.png"
      ],
      "identifier": "com.tw93.yacd",
      "active": true,
      "category": "DeveloperTool",
      "copyright": "",
      "deb": {
        "depends": [
          "libwebkit2gtk-4.0-dev",
          "build-essential",
          "curl",
          "wget",
          "libssl-dev",
          "libgtk-3-dev",
          "libayatana-appindicator3-dev",
          "librsvg2-dev",
          "gnome-video-effects",
          "gnome-video-effects-extra"
        ],
        "files": {
          "/usr/share/applications/com-tw93-yacd.desktop": "assets/com-tw93-yacd.desktop"
        }
      },
      "externalBin": [],
      "longDescription": "",
      "resources": [],
      "shortDescription": "",
      "targets": ["deb", "appimage"]
    }
  }
}
```

**Comment by Tlntin** [comment, marked as answer]
这个其实不是bug，如果你不动核心代码的话，是不会出现这个报错的。
相关的编译脚本在[这一行](https://github.com/tw93/Pake/blob/master/script/build.sh#LL48C5-L48C112)，以及[这一行](https://github.com/tw93/Pake/blob/master/script/build.sh#L59)
如果要修复，建议改这一行，将sd里面的固定替换改成正则模式替换（也可以直接改成你改动的package_name）。
只不过一般用户并不会去修改核心代码，所以这个影响几乎为零。

**Reply by Tlntin** [reply]
新版已经修复了。
